sw/host/opentitanlib/src/transport/hyperdebug/spi.rs

```diff
@@ -300,6 +351,189 @@ impl HyperdebugSpiTarget {
+        req.data[idx..idx + address_bytes.len()].clone_from_slice(address_bytes);
+        idx += address_bytes.len();
+        if cmd.get_address_width() == eeprom::DataWidth::Single
+            && cmd.get_data_width() == eeprom::DataWidth::Single
+            && cmd.get_dummy_cycles() % 8 == 0
+            && cmd.get_dummy_cycles() <= 32
+        {
+            // In cases when the number of dummy cycles is divisible by 8, stuff a number of
+            // zero bytes after the address.  This allows debuggers without native support for
+            // an arbitrary number of dummy cycles to perform the transaction.
+            req.flags |= (((cmd.get_dummy_cycles() / 8) as u32) << EEPROM_FLAGS_ALT_LEN_POS)
+                | (Self::width_encoding(eeprom::DataWidth::Single) << EEPROM_FLAGS_ALT_WIDTH_POS);
+            for _ in 0..cmd.get_dummy_cycles() / 8 {
+                req.data[idx] = 0x00;
+                idx += 1;
+            }
+        } else if cmd.get_dummy_cycles() < 32 {
+            req.flags |= (cmd.get_dummy_cycles() as u32) << EEPROM_FLAGS_DUMMY_CYCLES_POS
+        } else {
+            bail!(SpiError::InvalidDummyCycles(cmd.get_dummy_cycles()));
+        }
+        let data_start_offset = idx;
+        // Optional write data bytes
+        let databytes = std::cmp::min(USB_MAX_SIZE - 8 - data_start_offset, wbuf.len());
+        req.data[data_start_offset..data_start_offset + databytes]
+            .clone_from_slice(&wbuf[0..databytes]);
+        self.usb_write_bulk(&req.as_bytes()[0..8 + data_start_offset + databytes])?;
+        let mut index = databytes;
+
+        while index < wbuf.len() {
+            let mut req = CmdTransferContinue::new();
+            req.data_index = index as u16;
+            let databytes = std::cmp::min(USB_MAX_SIZE - 4, wbuf.len() - index);
+            req.data[0..databytes].clone_from_slice(&wbuf[index..index + databytes]);
+            self.usb_write_bulk(&req.as_bytes()[0..4 + databytes])?;
+            index += databytes;
+        }
+
+        let next_stream_state = if rbuf.len() == 0 {
+            StreamState::PendingWrite
+        } else {
+            StreamState::PendingRead(rbuf)
+        };
+        if double_buffer {
+            self.receive_streamed_data(stream_state)?;
+            Ok(next_stream_state)
+        } else {
+            self.receive_streamed_data(next_stream_state)?;
+            Ok(StreamState::NoPending)
+        }
+    }
+
+    fn receive_streamed_data(&self, stream_state: StreamState) -> Result<()> {
+        match stream_state {
+            StreamState::NoPending => self.receive_first_streaming(),
+            StreamState::PendingWrite => self.receive(&mut []),
+            StreamState::PendingRead(rbuf) => self.receive(rbuf),
+        }
+    }
+
+    fn eeprom_transmit_get_last_streamed_data(&self) -> Result<()> {
+        let mut req = CmdEepromTransferStart::new();
+        req.count = 0;
+        req.flags = 1 << 30;
```

Comment:
Replace this constant with a symbolic name.

Reply:
Done